src/main/java/org/jenkinsci/plugins/workflow/support/steps/StageStepExecution.java

```diff
@@ -78,226 +55,20 @@ public boolean start() throws Exception {
                     .start();
             return false;
         }
-        getContext().get(TaskListener.class).getLogger().println(Messages.StageStepExecution_non_block_mode_deprecated());
-        FlowNode node = getContext().get(FlowNode.class);
-        if (isInsideParallel(node)) {
-            throw new AbortException(Messages.StageStepExecution_the_stage_step_must_not_be_used_inside_a());
-        }
-        node.addAction(new LabelAction(step.name));
-        node.addAction(new StageActionImpl(step.name));
-        enter(getContext(), step.name, step.concurrency);
```

Comment:
Now considering just letting the build proceed with a warning _if_ `concurrency` was not being used. Would be compatible for everyone except those who were actually using the system superseded by `lock` + `milestone`, while still allowing the complicated code to be deleted.

Reply:
There is an error message above that probably needs to be changed to a better wording?
`throw new AbortException(Messages.StageStepExecution_concurrency_not_supported_in_block_mode());`

Reply:
I will review the current messages to ensure they still make sense.